ratis-server/src/main/java/org/apache/ratis/server/storage/FileChunkReader.java

```diff
@@ -65,7 +66,7 @@ public FileChunkReader(FileInfo info, RaftStorageDirectory directory) throws IOE
   public FileChunkProto readFileChunk(int chunkMaxSize) throws IOException {
     final long remaining = info.getFileSize() - offset;
     final int chunkLength = remaining < chunkMaxSize ? (int) remaining : chunkMaxSize;
-    final ByteString data = ByteString.readFrom(in, chunkLength);
```

Comment:
ByteString.readFrom(in, chunkLength) will read the whole file not just chunkLength